agent/agent.go

```diff
@@ -340,8 +342,11 @@ func (a *Agent) processNextTask(ctx context.Context, nt *apimodels.NextTaskRespo
 	}
 
 	if nt.TaskSecret == "" {
+		msg := "task response missing secret"
+		span.SetStatus(codes.Error, msg)
+		span.RecordError(errors.New(msg), trace.WithAttributes(attribute.String("task_id", tc.task.ID)))
```

Comment:
I've tended to use dot notation for attributes. I've found it enables discovery and helps keep them organized.

Reply:
Done.